Content.Server/Forensics/Systems/ForensicsSystem.cs

```diff
@@ -101,17 +104,51 @@ public void CopyForensicsFrom(ForensicsComponent src, EntityUid target)
             }
         }
 
-        private void OnAfterInteract(EntityUid uid, CleansForensicsComponent component, AfterInteractEvent args)
+        // Try: "Entity<CleansForensicsComponent> cleanitemcomp" or something!
+        private void OnAfterInteract(Entity<CleansForensicsComponent> cleanForensicsEntity, ref AfterInteractEvent args)
         {
             if (args.Handled || !args.CanReach)
                 return;
 
-            if (!TryComp<ForensicsComponent>(args.Target, out var forensicsComp))
+            args.Handled = StartClean(cleanForensicsEntity, args.User, args.Target);
+        }
+
+        private void OnUtilityVerb(Entity<CleansForensicsComponent> entity, ref GetVerbsEvent<UtilityVerb> args)
+        {
+            if (!args.CanInteract || !args.CanAccess || !args.CanInteract)
                 return;
 
-            if((forensicsComp.DNAs.Count > 0 && forensicsComp.CanDnaBeCleaned) || (forensicsComp.Fingerprints.Count + forensicsComp.Fibers.Count > 0))
+            // These need to be set outside for the anonymous method!
+            var user = args.User;
+            var target = args.Target;
+
+            var verb = new UtilityVerb()
             {
-                var doAfterArgs = new DoAfterArgs(EntityManager, args.User, component.CleanDelay, new CleanForensicsDoAfterEvent(), uid, target: args.Target, used: args.Used)
+                Act = () => StartClean(entity, user, target),
+                IconEntity = GetNetEntity(entity),
+                Text = Loc.GetString(Loc.GetString("forensics-verb-text")),
+                Message = Loc.GetString(Loc.GetString("forensics-verb-message"))
+            };
+
+            args.Verbs.Add(verb);
+        }
+
+        /// <summary>
+        ///     Attempts to clean the given item with the given CleansForensics entity.
+        /// </summary>
+        /// <param name="cleanForensicsEntity">The entity that is being used to clean the target.</param>
+        /// <param name="user">The user that is using the cleanForensicsEntity.</param>
+        /// <param name="target">The target of the forensics clean.</param>
+        /// <returns>True if the target can be cleaned and has some sort of DNA or fingerprints / fibers and false otherwise.</returns>
+        public bool StartClean(Entity<CleansForensicsComponent> cleanForensicsEntity, EntityUid user, EntityUid? target)
+        {
+            if (!TryComp<ForensicsComponent>(target, out var forensicsComp))
+                return false;
+
+            if ((forensicsComp.DNAs.Count > 0 && forensicsComp.CanDnaBeCleaned) || (forensicsComp.Fingerprints.Count + forensicsComp.Fibers.Count > 0))
```

Comment:
cant you merge the first part of the condition and the second part by adding the counts? also this should be a named var

Reply:
I personally don't think merging them is a good idea as I feel like its mixing two separate things and would make it less readable. If you think its the right path though I'll try to change it :)

I don't *quite* know what a named var is but I changed it a bit to be clearer!
Let me know if this is good 👍

```
var totalPrintsAndFibers = forensicsComp.Fingerprints.Count + forensicsComp.Fibers.Count;
var hasRemovableDNA = forensicsComp.DNAs.Count > 0 && forensicsComp.CanDnaBeCleaned;

if (hasRemovableDNA || totalPrintsAndFibers > 0)
```

Reply:
yeah thats a local var (which is what i meant), thats alright